packages/pocket-ic/BUILD.bazel

```
@@ -66,53 +67,48 @@ rust_test(
     deps = [":pocket-ic"] + DEPENDENCIES + TEST_DEPENDENCIES,
 )
 
-rust_test_suite(
+test_using_pocket_ic_server(
     name = "test",
     size = "small",
     srcs = ["tests/tests.rs"],
     data = [
         "tests/counter.wasm",
         "tests/icp_ledger.wasm",
         ":test_canister.wasm",
-        "//rs/pocket_ic_server:pocket-ic-server",
     ],
     env = {
-        "POCKET_IC_BIN": "$(rootpath //rs/pocket_ic_server:pocket-ic-server)",
         "COUNTER_WASM": "packages/pocket-ic/tests/counter.wasm",
         "LEDGER_WASM": "packages/pocket-ic/tests/icp_ledger.wasm",
         "TEST_WASM": "$(rootpath :test_canister.wasm)",
     },
+    # Override the default ["manual"] extra_HEAD_tags to ensure this test runs on CI.
+    extra_HEAD_tags = [],
     flaky = False,
+    macro = rust_test_suite,
     proc_macro_deps = MACRO_DEPENDENCIES,
     deps = [":pocket-ic"] + DEPENDENCIES + TEST_DEPENDENCIES,
 )
 
-rust_test_suite(
+test_using_pocket_ic_server(
     name = "restart",
     size = "medium",
     srcs = ["tests/restart.rs"],
-    data = [
-        "//rs/pocket_ic_server:pocket-ic-server",
-    ],
-    env = {
-        "POCKET_IC_BIN": "$(rootpath //rs/pocket_ic_server:pocket-ic-server)",
-    },
+    # Override the default ["manual"] extra_HEAD_tags to ensure this test runs on CI.
+    extra_HEAD_tags = [],
     flaky = False,
+    macro = rust_test_suite,
     proc_macro_deps = MACRO_DEPENDENCIES,
     deps = [":pocket-ic"] + DEPENDENCIES + TEST_DEPENDENCIES,
 )
 
-rust_test_suite(
+test_using_pocket_ic_server(
     name = "slow",
     size = "medium",
     srcs = ["tests/slow.rs"],
-    data = [
-        "//rs/pocket_ic_server:pocket-ic-server",
-    ],
-    env = {
-        "POCKET_IC_BIN": "$(rootpath //rs/pocket_ic_server:pocket-ic-server)",
-    },
+    # Override the default ["manual"] extra_HEAD_tags to ensure this test runs on CI.
+    extra_HEAD_tags = [],
```

```suggestion
    extra_HEAD_tags = [],
    # Override the default [] extra_mainnet_tags to skip this test on CI.
    extra_mainnet_tags = ["manual"]
```
Alternatively to https://github.com/dfinity/ic/pull/1468#pullrequestreview-2304084361, assuming that recent changes to the library (such as adding new functions calling features not yet available in the mainnet server) won't impact the NNS tests (that won't yet use those new functions until the HEAD server is released), we might skip this test against the mainnet server and verify the assumption on NNS tests by (manually, if this can't be automated) rerunning all tests that depend on the HEAD PocketIC library. Does that sounds more feasible than my linked suggestion to compile the library from mainnet sources?

---

That sounds very reasonable!